packages/runtime/test-runtime-utils/src/mocksForReconnection.ts

```diff
@@ -91,68 +91,83 @@ export class MockContainerRuntimeForReconnection extends MockContainerRuntime {
 		return -1;
 	}
 
-	public async initializeWithStashedOps(containerRuntime: MockContainerRuntimeForReconnection) {
+	public async initializeWithStashedOps(
+		fromContainerRuntime: MockContainerRuntimeForReconnection,
+	) {
 		if (this.pendingMessages.length !== 0 || this.clientSequenceNumber !== 0) {
 			throw new Error("applyStashedOps must be called first, and once.");
 		}
 
-		const { processedOps, pendingRemoteMessages } = containerRuntime;
-
-		if (processedOps === undefined) {
+		if (fromContainerRuntime.processedOps === undefined) {
 			throw new Error("containerRuntime must have trackRemoteOps true");
 		}
 
 		// shutdown the existing client
-		containerRuntime.flush();
-		containerRuntime.connected = false;
-		this.factory.removeContainerRuntime(containerRuntime);
+		fromContainerRuntime.connected = false;
+		fromContainerRuntime.flush();
+		this.factory.removeContainerRuntime(fromContainerRuntime);
+		// clear any unprocessed ops for this client
+		this.factory.clearOutstandingClientMessages(fromContainerRuntime.clientId);
 
 		// get the saved ops seen by the client, and its pending ops
-		const pendingMessages = containerRuntime.pendingMessages.splice(0);
-		const savedOps = [...processedOps.splice(0), ...pendingRemoteMessages.splice(0)];
+		const pendingMessages = fromContainerRuntime.pendingMessages.splice(0);
+		const remoteOps = [
+			...fromContainerRuntime.processedOps.splice(0),
+			...fromContainerRuntime.pendingRemoteMessages.splice(0),
```

Comment:
i'd expect this line to noop since the contents were already spliced out above.

Reply:
more meant--either `...fromContainerRuntime.pendingRemoteMessages.splice(0)` should be deleted, or you want to use `.slice` and not `.splice` on L113.

Reply:
i want the original pendingRemoteMessages to be emptied, slice won't empty the arry

Reply:
or is it that you didn't notice like 113 `pendingMessages` and line 116 `pendingRemoteMessages` are using different arrays

Reply:
oops. yeah it's that. mb :)